avocado/core/result.py

```diff
@@ -96,7 +100,7 @@ def __init__(self, job):
         """
         self.job_unique_id = getattr(job, "unique_id", None)
         self.logfile = getattr(job, "logfile", None)
-        self.tests_total = getattr(job.args, 'test_result_total', 1)
+        self.tests_total = 0
```

Comment:
It's hard for me to give a balanced feedback on this, because I have proposed conflicting changes.  Still, the general message that I'd like to send is that it's neither the Job alone or the test runner alone the task to come up with the number of tests.  I see that the definitive solution is to have proper remote test resolver and remove variant generation components.  Since this is a bit far off, we'll have to compromise.

Let's arrange a talk to coordinate this change with the result changes I'm also proposing.